zebra-network/src/constants.rs

```diff
@@ -8,10 +8,14 @@ use crate::protocol::external::types::*;
 use zebra_chain::parameters::NetworkUpgrade;
 
 /// The buffer size for the peer set.
+///
+/// We assume that Zebra nodes have at least 10 Mbps bandwidth. Therefore, a
+/// maximum-sized block will take 2 seconds to download. Based on the current
+/// `BLOCK_DOWNLOAD_TIMEOUT`, this is the largest buffer size we can support.
```

Comment:
Sorry, what's the connection between the buffer size and the block download timeout?

Reply:
I mean, the number 10 for the peer set buffer size was made up totally artificially (it should probably be 2 or 3 to avoid sender contention), so it's strange to see it justified in this way.

Reply:
> what's the connection between the buffer size and the block download timeout?

The timeout needs to be approximately the amount of time it takes to drain a full buffer, and one pending block request per peer, via the slowest network speed we support.

Otherwise, we can choke the local network connection with block downloads, leading to repeated failures.

I'll revise the comment and decrease the buffer size - I think this comment might actually belong in sync.rs.

Reply:
Fixed in 31e718a0ebad8ad32ae804790c54ba7b08e7d95d.

I decreased the buffer size to 3.

I also updated the sync timeout to 100, to match the default peer set size of 50, on a 10 Mbps node, downloading maximum-size blocks. (In practice, the actual blocks are smaller, but the calculation doesn't include latency. So I think they should balance out. Resets should also be rare, after this PR merges.)

I added some additional comments and documentation for network-constrained nodes.